backends-common/cassandra/src/main/java/org/apache/james/backends/cassandra/init/CassandraTypesProvider.java

```diff
@@ -0,0 +1,52 @@
+/****************************************************************
+ * Licensed to the Apache Software Foundation (ASF) under one   *
+ * or more contributor license agreements.  See the NOTICE file *
+ * distributed with this work for additional information        *
+ * regarding copyright ownership.  The ASF licenses this file   *
+ * to you under the Apache License, Version 2.0 (the            *
+ * "License"); you may not use this file except in compliance   *
+ * with the License.  You may obtain a copy of the License at   *
+ *                                                              *
+ *   http://www.apache.org/licenses/LICENSE-2.0                 *
+ *                                                              *
+ * Unless required by applicable law or agreed to in writing,   *
+ * software distributed under the License is distributed on an  *
+ * "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY       *
+ * KIND, either express or implied.  See the License for the    *
+ * specific language governing permissions and limitations      *
+ * under the License.                                           *
+ ****************************************************************/
+
+package org.apache.james.backends.cassandra.init;
+
+import java.util.Optional;
+import java.util.stream.Collectors;
+
+import com.datastax.driver.core.Session;
+import com.datastax.driver.core.UserType;
+import com.google.common.collect.ImmutableMap;
+import org.apache.james.backends.cassandra.components.CassandraModule;
+import org.apache.james.backends.cassandra.components.CassandraType;
+
+public class CassandraTypesProvider {
+    private final ImmutableMap<String, UserType> userTypes;
+
+    public CassandraTypesProvider(CassandraModule module, Session session) {
+        userTypes = module.moduleTypes()
+            .stream()
+            .collect(Collectors.collectingAndThen(
+                Collectors.toMap(
```

Comment:
we should write a ImmutableMapCollector (see http://www.jayway.com/2014/09/29/java-8-collector-for-gauvas-linkedhashmultimap/ )